tests/unit/browser/webkit/network/test_pac.py

```diff
@@ -0,0 +1,235 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+import http.server
+import threading
+import logging
+
+from PyQt5.QtCore import QUrl, QDateTime, QDate, QTime, Qt
+from PyQt5.QtNetwork import QNetworkProxy, QNetworkProxyQuery
+
+from qutebrowser.browser.webkit.network import pac
+
```

Comment:
_Very_ strange because `isPlainHostName` doesn't use anything from Qt at all (excluding signals/slots)! I'm okay with this change but it's interesting what's the root cause...